aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -44,9 +71,26 @@ struct aesd_buffer_entry *aesd_circular_buffer_find_entry_offset_for_fpos(struct
 */
 void aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const struct aesd_buffer_entry *add_entry)
 {
-    /**
-    * TODO: implement per description
-    */
+	if (buffer == NULL || add_entry == NULL) {
+        	return;
+    	}
+    // Overwrite the oldest entry if the buffer is full
+    if (buffer->full) {
+        buffer->out_offs = (buffer->out_offs + 1) % AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED;
+    }
+
+    // Add new entry at the current input position
+    buffer->entry[buffer->in_offs] = *add_entry;
+
+    // Advance the input position
+    buffer->in_offs = (buffer->in_offs + 1) % AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED;
+
+    // Check if we've wrapped around and set the full flag if necessary
+    if (buffer->in_offs == buffer->out_offs) {
+        buffer->full = true;
+    } else {
+        buffer->full = false;
```

Comment:
@shruthi-thallapally I don't think you will ever add an entry and go from buffer full to not full, so this should be dead code.